server/src/main/java/org/candlepin/controller/CandlepinPoolManager.java

```diff
@@ -177,13 +177,16 @@ void refreshPoolsWithRegeneration(Owner owner, boolean lazy) {
         // remove everything that isn't actually active
         subIds.removeAll(deletedSubs);
         // delete pools whose subscription disappeared:
-
+        List<Pool> poolsToDelete = new ArrayList<Pool>();
         for (Pool p : poolCurator.getPoolsFromBadSubs(owner, subIds)) {
             if ((p.getType() == PoolType.NORMAL || p.getType() == PoolType.BONUS) &&
                 (ueberPoolId == null || !ueberPoolId.equals(p.getId()))) {
-                deletePool(p);
+                poolsToDelete.add(p);
             }
         }
+        if (!poolsToDelete.isEmpty()) {
+            deletePools(poolsToDelete);
```

Comment:
Delete pools has an empty do nothing check -- up to yourself, but we could remove the if here.